### PR TITLE
Prepare v0.45.0 release

### PR DESCRIPTION
**Pull request by philipaconrad**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -3,7 +3,161 @@
...
+Rego equivalent.
+
+### Tooling, SDK, and Runtime
+
+- cli: Fix doubled CLI hints/errors. ([#5115](https://github.com/open-policy-agent/opa/issues/5115)) authored by @ivanphdz
+- cli/test: Add capabilities flag to test command. (authored by @ivanphdz)
+- fmt: Fix blank lines after multiline expressions. (authored by @jaspervdj)
+- internal/report: Include heap usage in the telemetry report.
+- plugins/logs: Improve error message when decision log chunk size is greater than the upload limit. ([#5155](https://github.com/open-policy-agent/opa/issues/5155))
+- ir: Make golang code public.
```
Comment (ashutosh-narkar) [inline CHANGELOG.md:107]: Nit: Can we add some context here.

File: CHANGELOG.md [hunk 2]
```diff
@@ -3,7 +3,161 @@
...
+- ast/parser+formatter: Allow 'if' in rule 'else' statements.
+- ast/schema: Add support for recursive json schema elements. ([#5166](https://github.com/open-policy-agent/opa/issues/5166)) authored and reported by @liamg
+- ast/schema: Fix race condition in parsing with reused references.(authored by @liamg)
+- internal/gojsonschema: Fix race condition in `SetAllowNet`. ([#5187](https://github.com/open-policy-agent/opa/issues/5187)) authored and reported by @liamg
+- ast/compiler: Rewrite declared variables in function calls and recursively rewrite local variables in `with` clauses. ([#5148](https://github.com/open-policy-agent/opa/issues/5148)) authored and reported by @liu-du
+- ast: Skip rules when parsing a body (or query) to help improve ambiguous parsing cases.
+
+### Topdown
+
+- topdown/regex: Add `regex.replace(s, p, v)` to builtins.
```
Comment (ashutosh-narkar) [inline CHANGELOG.md:120]: Can be omitted since we already call it out at the top.

**Review by ashutosh-narkar — APPROVED** [review]

File: CHANGELOG.md (hunk 2 above)
Comment (philipaconrad) [reply CHANGELOG.md:120]: Sounds good. Will edit.

File: CHANGELOG.md (hunk 1 above)
Comment (philipaconrad) [reply CHANGELOG.md]: Done. :+1:

**Comment by philipaconrad** [comment]
Will merge when green.

File: CHANGELOG.md [hunk 3]
```diff
@@ -3,7 +3,160 @@
...
+an opt-in system for caching the inputs and outputs of these built-ins
+during policy evaluation, and can include this information in decision
+log entries.
+
+A new top-level config key is used to enable the non-deterministic
+builtin caching feature, as shown below:
+
+    nd_builtin_cache: true
+
+This data is exposed to OPA's [decision log masking system](https://www.openpolicyagent.org/docs/latest/management-decision-logs/#masking-sensitive-data)
```
Comment (srenatus) [inline CHANGELOG.md:25]: [nit] lets replace latest with v0.45.0 here -- whatever happens in the future to that page, this link would stay intact.

File: CHANGELOG.md [hunk 4]
```diff
@@ -3,7 +3,160 @@
...
+- ast/parser+formatter: Allow 'if' in rule 'else' statements.
+- ast/schema: Add support for recursive json schema elements. ([#5166](https://github.com/open-policy-agent/opa/issues/5166)) authored and reported by @liamg
+- ast/schema: Fix race condition in parsing with reused references.(authored by @liamg)
+- internal/gojsonschema: Fix race condition in `SetAllowNet`. ([#5187](https://github.com/open-policy-agent/opa/issues/5187)) authored and reported by @liamg
+- ast/compiler: Rewrite declared variables in function calls and recursively rewrite local variables in `with` clauses. ([#5148](https://github.com/open-policy-agent/opa/issues/5148)) authored and reported by @liu-du
+- ast: Skip rules when parsing a body (or query) to help improve ambiguous parsing cases.
+
+### Topdown
+
+- topdown/object: Rework `object.union_n` to use in-place merge algorithm. (reported by @charlesdaniels)
```
Comment (srenatus) [inline CHANGELOG.md:120]: OK This really is nitpicking with a magnifier and I'm not even sure what the style guide says about this, _but_ I'd drop the period there:
```suggestion
- topdown/object: Rework `object.union_n` to use in-place merge algorithm (reported by @charlesdaniels)
```
(Feel free to ignore. 🙃)

Anyhow, you're consistent, so it's OK to keep as-is.

File: CHANGELOG.md [hunk 5]
```diff
@@ -3,7 +3,160 @@
...
+
+- website/security: Style improvements. (authored by @orweis)
+
+### Miscellaneous
+
+- ci: Add `prealloc` linter check and linter fixes.
+- ci: Add govulncheck to Nightly CI.
+- build/wasm: Use golang1.16 `go:embed` mechanism.
+- util/backoff: Seed from math/rand source.
+- trivyignore: Add wasmtime-go CVE-2022-31169.
```
Comment (srenatus) [inline CHANGELOG.md:151]: This was removed. We no longer need to ignore the vuln, because we're no longer using a vulnerable version.

**Review by srenatus — DISMISSED** [review]
Thank you!

File: CHANGELOG.md (hunk 3 above)
Comment (philipaconrad) [reply CHANGELOG.md:25]: Rolling this into the CHANGELOG. :+1:

File: CHANGELOG.md (hunk 5 above)
Comment (philipaconrad) [reply CHANGELOG.md:151]: Aha! I'll remove that wasmtime-go line then.

File: CHANGELOG.md (hunk 4 above)
Comment (philipaconrad) [reply CHANGELOG.md:120]: Keeping as-is for now, because if we want to refactor CHANGELOG entries later, at least it's consistent.

**Review by ashutosh-narkar — APPROVED** [review]

**Review by srenatus — APPROVED** [review]
🥳

**Comment by philipaconrad** [comment]
Overriding to merge before the perf tests are done (they've gone green 2-3 times already, so I think we're good).

**Pull request merged**